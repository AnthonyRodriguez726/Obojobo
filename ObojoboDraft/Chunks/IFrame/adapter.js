import Common from 'Common'

import IFrameMediaTypes from './iframe-media-types'
import IFrameFitTypes from './iframe-fit-types'
import IFrameControlTypes from './iframe-control-types'

const setModelStateProp = Common.util.setModelStateProp
const cloneProps = Common.util.cloneProps
const propsList = [
	'type',
	'src',
	'width',
	'height',
	'zoom',
	'border',
	'autoload',
	'fit',
	'title',
	'controls'
]

export default {
	construct(model, attrs) {
		const s = model.setModelStateProp.bind(model)

		s('type', IFrameMediaTypes.MEDIA, p => p.toLowerCase(), [
			IFrameMediaTypes.WEBPAGE,
			IFrameMediaTypes.MEDIA
		])

		let defaultBorder
		let defaultFit
		let defaultControls

		switch (model.modelState.type) {
			case IFrameMediaTypes.WEBPAGE:
				defaultBorder = true
				defaultFit = IFrameFitTypes.SCROLL
				defaultControls = [
					IFrameControlTypes.ZOOM,
					IFrameControlTypes.RELOAD,
					IFrameControlTypes.NEW_WINDOW
				]
				break

			case IFrameMediaTypes.MEDIA:
			default:
				defaultBorder = false
				defaultFit = IFrameFitTypes.SCALE
				defaultControls = [IFrameControlTypes.RELOAD]
				break
		}

		s('border', defaultBorder)
		s('fit', defaultFit, p => p.toLowerCase(), [IFrameFitTypes.SCROLL, IFrameFitTypes.SCALE])
		s('src', null)
		s('width', null, p => parseInt(p, 10) || null)
		s('height', null, p => parseInt(p, 10) || null)
		s('initialZoom', 1, p => parseFloat(p) || 1)
		s('autoload', false, p => p === true)
		s('title', null)
		s('controls', defaultControls, p => p.split(',').map(c => c.toLowerCase().replace(/ /g, '')))
	},

	clone(model, clone) {
		cloneProps(clone.modelState, model.modelState, propsList)
	},

	toJSON(model, json) {
		cloneProps(json.content, model.modelState, propsList)
	},

	toText(model) {
		return model.modelState.src
	}
}
