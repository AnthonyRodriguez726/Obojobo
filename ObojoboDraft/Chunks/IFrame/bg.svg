<svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 38.16 38.16">
	<polygon fill-opacity="0.06" points="1.42 0 0 1.42 0 0 1.42 0"/>
	<polygon fill-opacity="0.06" points="38.16 0 38.16 1.42 1.42 38.16 0 38.16 38.16 0"/>
	<polygon fill-opacity="0.06" points="8.01 0 0 8.01 0 9.42 9.42 0 8.01 0"/>
	<polygon fill-opacity="0.06" points="38.16 8 38.16 9.42 9.42 38.16 8 38.16 38.16 8"/>
</svg>