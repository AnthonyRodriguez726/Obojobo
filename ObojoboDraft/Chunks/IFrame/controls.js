import './controls.scss'

import parseURL from 'url-parse'

import Common from 'Common'
let isOrNot = Common.util.isOrNot

export default props => {
	let controlsOpts = props.controlsOptions
	let isShowingSizeControls = controlsOpts.zoom || controlsOpts.expand || controlsOpts.unexpand
	let newWindowToolTip = parseURL(props.newWindowSrc).hostname

	return (
		<div
			className={
				'obojobo-draft--chunks--iframe--controls' +
				isOrNot(controlsOpts.unexpand, 'unexpandable') +
				isOrNot(controlsOpts.expand, 'expandable') +
				isOrNot(controlsOpts.zoom, 'zoomable') +
				isOrNot(controlsOpts.reload, 'reloadable')
			}
		>
			{controlsOpts.reload ? (
				<div className="control-button-container reload">
					<button className="reload-button" onClick={props.reload}>
						Reload
					</button>
					<span className="tool-tip">Reload</span>
				</div>
			) : null}
			{controlsOpts.newWindow ? (
				<div className="new-window-link">
					<a target="_blank" href={props.newWindowSrc}>
						View in a new window
					</a>

					<span className="tool-tip">{newWindowToolTip}</span>
				</div>
			) : null}
			{isShowingSizeControls ? (
				<div className="size-controls">
					{controlsOpts.zoom ? (
						<div className="zoom-controls">
							{props.isZoomAbleToBeReset ? (
								<div className="control-button-container zoom-reset">
									<button className="zoom-reset-button" onClick={props.zoomReset}>
										Reset zoom
									</button>
									<span className="tool-tip">Reset zoom</span>
								</div>
							) : null}
							<div className="control-button-container zoom-out">
								<button
									disabled={props.isUnableToZoomOut}
									className="zoom-out-button"
									onClick={props.zoomOut}
								>
									Zoom out
								</button>
								<span className="tool-tip">
									{props.isUnableToZoomOut ? "Whoa that's tiny! ðŸ˜²" : 'Zoom out'}
								</span>
							</div>
							<div className="control-button-container zoom-in">
								<button className="zoom-in-button" onClick={props.zoomIn}>
									Zoom in
								</button>
								<span className="tool-tip">Zoom in</span>
							</div>
						</div>
					) : null}
					{controlsOpts.expand ? (
						<div className="control-button-container expand">
							<button className="expand-button" onClick={props.expand}>
								View larger
							</button>
							<span className="tool-tip">View larger</span>
						</div>
					) : null}
					{controlsOpts.unexpand ? (
						<div className="control-button-container unexpand">
							<button className="unexpand-button" onClick={props.expandClose}>
								Resume to original size
							</button>
							<span className="tool-tip">Resume to original size</span>
						</div>
					) : null}
				</div>
			) : null}
		</div>
	)
}
