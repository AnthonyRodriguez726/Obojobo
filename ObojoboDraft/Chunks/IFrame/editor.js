import React from 'react'

const IFRAME_NODE = 'ObojoboDraft.Chunks.IFrame'

class Node extends React.Component {
	constructor(props) {
		super(props);
		this.state = this.props.node.data.get('content')
	}

	handleTypeChange(event){
		const editor = this.props.editor
		const change = editor.value.change()

		this.setState({ src: event.target.value })

		change.setNodeByKey(this.props.node.key, { data: { content: {
			src: event.target.value
		}}})
		editor.onChange(change)
	}

	renderInput() {
		return (
			<div>
				<input
					value={this.state.src}
					onChange={event => this.handleSourceChange(event)}
					onClick={event => event.stopPropagation()} />
			</div>
		)
	}

	renderFrame() {
		const { isFocused } = this.props

		const wrapperStyle = {
			position: 'relative',
			outline: isFocused ? '2px solid blue' : 'none',
		}

		const maskStyle = {
			display: isFocused ? 'none' : 'block',
			position: 'absolute',
			top: '0',
			left: '0',
			height: '100%',
			width: '100%',
			cursor: 'cell',
			zIndex: 1,
		}

		const iframeStyle = {
			display: 'block',
		}

		return (
			<div className={'obojobo-draft--chunks--iframe viewer'} style={wrapperStyle}>
				<div style={maskStyle} />
				<iframe
					is
					src={this.state.src}
					frameBorder="0"
					allowFullScreen="true"
				/>
			</div>
		)
	}

	render() {
		const { isSelected } = this.props

		return (
			<div className={'component'} {...this.props.attributes}>
				{this.renderFrame()}
				{isSelected ? this.renderInput() : null}
			</div>
		)
	}
}

const insertNode = change => {
	change
		.insertBlock({
			type: IFRAME_NODE,
			data: { content: { src: 'https://www.youtube.com/embed/dQw4w9WgXcQ' } },
			isVoid: true
		})
		.collapseToStartOfNextText()
		.focus()
}

const slateToObo = node => {
	const json = {}
	json.id = node.key
	json.type = node.type
	json.content = node.data.get('content') || {}

	return json
}

const plugins = {
	renderNode(props) {
		switch (props.node.type) {
			case IFRAME_NODE:
				return <Node {...props} />
		}
	},
	schema: {
		blocks: {
			'ObojoboDraft.Chunks.IFrame': {
				isVoid: true,
			}
		}
	}
}

const IFrame = {
	components: {
		Node,
	},
	helpers: {
		insertNode,
		slateToObo
	},
	plugins
}

export default IFrame
