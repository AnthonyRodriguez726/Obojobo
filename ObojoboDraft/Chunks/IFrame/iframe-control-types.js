export default {
	ZOOM: 'zoom',
	RELOAD: 'reload'
}
