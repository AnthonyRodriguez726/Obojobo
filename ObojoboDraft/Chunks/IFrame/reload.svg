<svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 14.96 13">
	<defs>
		<style>.cls-1{fill:#b497d3;}</style>
	</defs>
	<title>reload</title>
	<path class="cls-1" d="M9.5,2.74a6.48,6.48,0,0,0-6.41,5H1.35a.41.41,0,0,0-.29.69L3.62,11a.4.4,0,0,0,.57,0L6.75,8.43a.41.41,0,0,0-.29-.69H4.69a5,5,0,1,1,1.17,5l-1.36.78A6.49,6.49,0,0,0,15.9,9.15,6.58,6.58,0,0,0,9.5,2.74Z" transform="translate(-0.94 -2.74)"/>
</svg>