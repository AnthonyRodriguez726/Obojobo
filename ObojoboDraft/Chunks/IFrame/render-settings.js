import Viewer from 'Viewer'
let MediaUtil = Viewer.util.MediaUtil

let SIZE_STATE_EXPANDED = 'expanded'
let SIZE_STATE_ABLE_TO_EXPAND = 'ableToExpand'
let SIZE_STATE_UNABLE_TO_EXPAND = 'unableToExpand'

let getIsShowing = (mediaState, model) => {
	return (
		(model.modelState.autoload || MediaUtil.isShowingMedia(mediaState, model)) &&
		model.modelState.src !== null
	)
}

let getControlsOptions = (modelState, sizeState) => {
	let isZoomControlEnabled = modelState.controls.indexOf('zoom') > -1
	let isReloadControlEnabled = modelState.controls.indexOf('reload') > -1
	let isExpandControlNeeded =
		modelState.controls.indexOf('expand') > -1 && sizeState === SIZE_STATE_ABLE_TO_EXPAND
	let isUnexpandControlNeeded = sizeState === SIZE_STATE_EXPANDED
	let newWindowEnabled = modelState.newWindow !== null

	return {
		zoom: isZoomControlEnabled,
		reload: isReloadControlEnabled,
		expand: isExpandControlNeeded,
		unexpand: isUnexpandControlNeeded,
		newWindow: newWindowEnabled,
		isControlsEnabled:
			isZoomControlEnabled ||
			isReloadControlEnabled ||
			isExpandControlNeeded ||
			isUnexpandControlNeeded ||
			newWindowEnabled
	}
}

let getSizeState = (expandedSize, scaleAmount, mediaSize) => {
	if (mediaSize === 'large') return SIZE_STATE_EXPANDED
	if (expandedSize === 'full' || (expandedSize === 'restricted' && scaleAmount < 1))
		return SIZE_STATE_ABLE_TO_EXPAND
	return SIZE_STATE_UNABLE_TO_EXPAND
}

let getMediaSize = (mediaState, model, defaultSizeIfNotSet) => {
	return MediaUtil.getSize(mediaState, model) || defaultSizeIfNotSet
}

let getDisplayedTitle = modelState => {
	if (modelState.src === null) {
		return 'IFrame missing src attribute'
	} else if (modelState.title) {
		return modelState.title
	}

	return (modelState.src || '').replace(/^https?:\/\//, '')
}

let getSetDimensions = (modelState, defaultWidth, defaultHeight) => {
	return {
		w: modelState.width || defaultWidth,
		h: modelState.height || defaultHeight
	}
}

let getScaleAmount = (actualWidth, padding, setWidth) => {
	return Math.min(1, (actualWidth - padding) / setWidth)
}

let getScaleDimensions = (modelState, zoom, isExpanded, scaleAmount, minScale, setDimensions) => {
	let scale
	let containerStyle = {}

	if (isExpanded) {
		scale = zoom

		if (modelState.expandedSize === 'restricted') {
			containerStyle.maxWidth = setDimensions.w
			containerStyle.maxHeight = setDimensions.h
		}
	} else {
		if (modelState.fit === 'scroll') {
			scale = zoom
			containerStyle = {
				width: setDimensions.w,
				height: setDimensions.h
			}
		} else {
			scale = scaleAmount * zoom
			containerStyle = {
				width: setDimensions.w
			}
		}
	}

	scale = Math.max(minScale, scale)

	return {
		scale,
		containerStyle
	}
}

let getIFrameStyle = scale => {
	return {
		transform: `scale(${scale})`,
		width: 1 / scale * 100 + '%',
		height: 1 / scale * 100 + '%'
	}
}

let getAfterStyle = (setWidth, setHeight, fit) => {
	return fit === 'scale'
		? {
				paddingTop: setHeight / setWidth * 100 + '%'
		  }
		: {
				height: setHeight
		  }
}

let getZoomValues = (mediaState, model) => {
	let userZoom = MediaUtil.getZoom(mediaState, model)
	let initialZoom = model.modelState.zoom
	let currentZoom = userZoom || initialZoom
	let isZoomDifferentFromInitial = currentZoom !== initialZoom

	return {
		userZoom,
		initialZoom,
		currentZoom,
		isZoomDifferentFromInitial
	}
}

let getRenderSettings = (
	model,
	actualWidth,
	padding,
	defaultSize,
	defaultWidth,
	defaultHeight,
	minScale,
	mediaState
) => {
	let ms = model.modelState
	let zoomValues = getZoomValues(mediaState, model)
	let zoom = zoomValues.currentZoom
	let setDimensions = getSetDimensions(ms, defaultWidth, defaultHeight)
	let mediaSize = getMediaSize(mediaState, model, defaultSize)
	let scaleAmount = getScaleAmount(actualWidth, padding, setDimensions.w)
	let displayedTitle = getDisplayedTitle(ms)
	let sizeState = getSizeState(ms.expandedSize, scaleAmount, mediaSize)
	let isExpanded = sizeState === SIZE_STATE_EXPANDED
	let scaleDimensions = getScaleDimensions(
		ms,
		zoom,
		isExpanded,
		scaleAmount,
		minScale,
		setDimensions
	)
	let controlsOpts = getControlsOptions(ms, sizeState)
	let isAtMinScale = scaleDimensions.scale === minScale
	let iframeStyle = getIFrameStyle(scaleDimensions.scale)
	let afterStyle = getAfterStyle(setDimensions.w, setDimensions.h, ms.fit)
	let isShowing = getIsShowing(mediaState, model)

	return {
		zoomValues,
		zoom,
		mediaSize,
		displayedTitle,
		isExpanded,
		scaleDimensions,
		isShowing,
		controlsOpts,
		isAtMinScale,
		iframeStyle,
		afterStyle
	}
}

export {
	getIsShowing,
	getControlsOptions,
	getSizeState,
	getMediaSize,
	getDisplayedTitle,
	getSetDimensions,
	getScaleAmount,
	getScaleDimensions,
	getIFrameStyle,
	getAfterStyle,
	getZoomValues,
	getRenderSettings
}
