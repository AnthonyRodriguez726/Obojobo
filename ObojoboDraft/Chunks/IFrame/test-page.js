const systemFont =
	'-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol"'

const getBoxStyle = size => {
	return {
		outline: '1px solid #00bcd4',
		position: 'absolute',
		left: 0,
		right: 0,
		display: 'block',
		width: size,
		height: size
	}
}

const getSpanStyle = size => {
	return {
		fontFamily: systemFont,
		position: 'absolute',
		right: 0,
		bottom: 0,
		color: 'white',
		background: '#00bcd4'
	}
}

export default class TestPage extends React.Component {
	render() {
		return (
			<body
				style={{
					padding: 0,
					margin: 0,
					width: 1600,
					height: 1600,
					backgroundSize: '10px 10px',
					backgroundColor: '#f4f4f4',
					backgroundImage:
						'linear-gradient(to right, #e6e6e6 1px, transparent 1px), linear-gradient(to bottom, #e6e6e6 1px, transparent 1px)'
				}}
			>
				<div style={getBoxStyle(50)}>
					<span style={getSpanStyle(50)}>50</span>
				</div>
				<div style={getBoxStyle(100)}>
					<span style={getSpanStyle(100)}>100</span>
				</div>
				<div style={getBoxStyle(200)}>
					<span style={getSpanStyle(200)}>200</span>
				</div>
				<div style={getBoxStyle(400)}>
					<span style={getSpanStyle(400)}>400</span>
				</div>
				<div style={getBoxStyle(800)}>
					<span style={getSpanStyle(800)}>800</span>
				</div>
				<div style={getBoxStyle(1600)}>
					<span style={getSpanStyle(1600)}>1600</span>
				</div>
				<div
					style={{
						position: 'fixed',
						left: '50%',
						top: '50%',
						margin: '0',
						background: 'rgba(0, 0, 0, 0.5)',
						color: 'white',
						padding: '0.5em 1.5em',
						borderRadius: '4em',
						transform: 'translate(-50%, -50%)',
						fontWeight: 'bold',
						textShadow: '0 1px rgba(0, 0, 0, 0.5)',
						fontFamily: systemFont,
						textAlign: 'center',
						pointerEvents: 'none'
					}}
				>
					Iframe Test Page
				</div>
				<div
					style={{
						position: 'fixed',
						bottom: 0,
						fontSize: 14,
						left: '50%',
						margin: 0,
						background: 'rgb(223, 246, 249)',
						padding: '0.5em 1.5em',
						transform: 'translate(-50%, 0)',
						fontFamily: systemFont,
						color: '#007a8a',
						width: '60%',
						textAlign: 'center',
						pointerEvents: 'none',
						borderRadius: '0.25em',
						borderBottomLeftRadius: 0,
						borderBottomRightRadius: 0
					}}
				>
					You're seeing this test page because no <code style={{ fontWeight: 'bold' }}>src</code>{' '}
					attribute was set. You can use this test page to see how this iframe displays. Set the{' '}
					<code style={{ fontWeight: 'bold' }}>src</code> attribute to a URL to show that page
					instead.
				</div>
			</body>
		)
	}
}
