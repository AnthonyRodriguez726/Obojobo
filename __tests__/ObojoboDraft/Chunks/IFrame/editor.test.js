import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import IFrame from '../../../../ObojoboDraft/Chunks/IFrame/editor'
const IFRAME_NODE = 'ObojoboDraft.Chunks.IFrame'

describe('IFrame editor', () => {
	test('Node component', () => {
		const Node = IFrame.components.Node
		const component = renderer.create(
			<Node
				attributes={{ dummy: 'dummyData' }}
				children={'mockChildren'}
				node={{
					data: {
						get: () => {
							return {}
						}
					}
				}}
			/>
		)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('Node component changes input', () => {
		const Node = IFrame.components.Node

		const change = {
			setNodeByKey: jest.fn()
		}

		const component = mount(
			<Node
				children={'mockChildren'}
				node={{
					data: {
						get: () => {
							return {}
						}
					}
				}}
				isFocused={true}
				isSelected={true}
				editor={{
					value: { change: () => change },
					onChange: jest.fn()
				}}
			/>
		)
		const tree = component.html()

		const click = component.find('input').simulate('click', {
			stopPropagation: () => true
		})

		const click2 = component.find('input').simulate('change', {
			target: { value: 'mockInput' }
		})

		expect(tree).toMatchSnapshot()
	})

	test('insertNode calls change methods', () => {
		const change = {}
		change.insertBlock = jest.fn().mockReturnValueOnce(change)
		change.collapseToStartOfNextText = jest.fn().mockReturnValueOnce(change)
		change.focus = jest.fn().mockReturnValueOnce(change)

		IFrame.helpers.insertNode(change)

		expect(change.insertBlock).toHaveBeenCalled()
		expect(change.collapseToStartOfNextText).toHaveBeenCalled()
		expect(change.focus).toHaveBeenCalled()
	})

	test('slateToObo converts a Slate node to an OboNode with content', () => {
		const slateNode = {
			key: 'mockKey',
			type: 'mockType',
			data: {
				get: type => {
					return null
				}
			},
			text: 'mockText'
		}
		const oboNode = IFrame.helpers.slateToObo(slateNode)

		expect(oboNode).toMatchSnapshot()
	})

	test('oboToSlate converts an OboNode to a Slate node', () => {
		const oboNode = {
			id: 'mockKey',
			type: 'mockType',
			content: { width: 'large' }
		}
		const slateNode = IFrame.helpers.oboToSlate(oboNode)

		expect(slateNode).toMatchSnapshot()
	})

	test('oboToSlate converts an OboNode to a Slate node with a caption', () => {
		const oboNode = {
			id: 'mockKey',
			type: 'mockType',
			content: {}
		}
		const slateNode = IFrame.helpers.oboToSlate(oboNode)

		expect(slateNode).toMatchSnapshot()
	})

	test('plugins.renderNode renders a button when passed', () => {
		const props = {
			node: {
				type: IFRAME_NODE,
				data: {
					get: () => {
						return {}
					}
				}
			}
		}

		expect(IFrame.plugins.renderNode(props)).toMatchSnapshot()
	})
})
