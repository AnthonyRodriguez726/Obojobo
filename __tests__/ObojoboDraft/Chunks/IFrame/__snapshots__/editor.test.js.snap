// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IFrame editor Node builds the expected component 1`] = `
<div
  className="component"
  dummy="dummyData"
>
  <div
    className="obojobo-draft--chunks--iframe viewer"
    style={
      Object {
        "outline": "none",
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "cursor": "cell",
          "display": "block",
          "height": "100%",
          "left": "0",
          "position": "absolute",
          "top": "0",
          "width": "100%",
          "zIndex": 1,
        }
      }
    />
    <iframe
      allowFullScreen="true"
      frameBorder="0"
      is={true}
      src={undefined}
    />
  </div>
</div>
`;

exports[`IFrame editor Node component 1`] = `
<div
  className="component"
  dummy="dummyData"
>
  <div
    className="obojobo-draft--chunks--iframe viewer"
    style={
      Object {
        "outline": "none",
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "cursor": "cell",
          "display": "block",
          "height": "100%",
          "left": "0",
          "position": "absolute",
          "top": "0",
          "width": "100%",
          "zIndex": 1,
        }
      }
    />
    <iframe
      allowFullScreen="true"
      frameBorder="0"
      is={true}
      src={undefined}
    />
  </div>
</div>
`;

exports[`IFrame editor oboToSlate converts an OboNode to a Slate node 1`] = `
Object {
  "data": Object {
    "content": Object {
      "width": "large",
    },
  },
  "isVoid": true,
  "key": "mockKey",
  "object": "block",
  "type": "mockType",
}
`;

exports[`IFrame editor oboToSlate converts an OboNode to a Slate node with a caption 1`] = `
Object {
  "data": Object {
    "content": Object {},
  },
  "isVoid": true,
  "key": "mockKey",
  "object": "block",
  "type": "mockType",
}
`;

exports[`IFrame editor plugins.renderNode renders a button when passed 1`] = `
<Node
  node={
    Object {
      "data": Object {
        "get": [Function],
      },
      "type": "ObojoboDraft.Chunks.IFrame",
    }
  }
/>
`;

exports[`IFrame editor slateToObo converts a Slate node to an OboNode with content 1`] = `
Object {
  "children": Array [],
  "content": Object {},
  "id": "mockKey",
  "type": "mockType",
}
`;
